### DISCUSSION TITLE
Looking for a way to run custom code before any test class gets instantiated

### DISCUSSION THREAD

**Discussion by fteotini** [xunit/xunit#2287] [Question] [answered]
HI,
As per title, I'm trying to execute some boilerplate code before any test class gets instantiated or run.

I know of Collection attribute, but this makes serialized all the tests using it.
I could create a base class with some logic to handle just une code run even if it gets extended by multiple test classes, but I don't want to preclude these from inheriting when appropriate.

There exists some method, like custom class attribute or any other dark magic to achieve what I'm looking for?
Thanks in advance

**Comment by bradwilson** [comment, marked as answer]
In v2, there's nothing built in, but you can use this third party package to get an assembly-level fixtures, which would allow you to run code before any tests run, and after all tests have finished: https://www.nuget.org/packages/xunit.assemblyfixture/

This is planned to be built into v3, through the use of assembly-level `BeforeAfterTestAttribute`: https://github.com/xunit/xunit/blob/main/src/xunit.v3.core/Sdk/BeforeAfterTestAttribute.cs#L11
